src/components/toolbar/toolbarFieldDisplayName.js

```diff
@@ -87,72 +108,50 @@ const ToolbarFieldDisplayName = ({
-
-  /**
-   * On enter submit value, on type submit value, and on esc ignore (clear value at component level).
-   *
-   * @event onKeyUp
-   * @param {object} event
-   */
-  const onKeyUp = event => {
-    switch (event.keyCode) {
-      case 13:
-        onSubmit(event.value);
-        break;
-      case 27:
-        break;
-      default:
-        debounced(event.value);
-        break;
-    }
-  };
+/**
+ * ToDo: evaluate the debounce milliseconds, currently based off platforms default 800 ms
+ */
+/**
+ * Display a display name input field for search.
+ *
+ * @param {object} props
+ * @param {translate} [props.t=translate]
+ * @param {useProductInventoryHostsQuery} [props.useProductInventoryHostsQuery=useProductInventoryHostsQuery]
+ * @param {useOnClear} [props.useOnClear=useOnClear]
+ * @param {useOnKeyUp} [props.useOnKeyUp=useOnKeyUp]
+ * @fires onClear
+ * @fires onKeyUp
+ * @returns {JSX.Element}
+ */
+const ToolbarFieldDisplayName = ({
+  t = translate,
+  useProductInventoryHostsQuery: useAliasProductInventoryHostsQuery = useProductInventoryHostsQuery,
+  useOnClear: useAliasOnClear = useOnClear,
+  useOnKeyUp: useAliasOnKeyUp = useOnKeyUp
+}) => {
+  const { [RHSM_API_QUERY_SET_TYPES.DISPLAY_NAME]: currentValue } = useAliasProductInventoryHostsQuery();
+  const onClear = useAliasOnClear();
+  const onKeyUp = useAliasOnKeyUp();
 
   return (
     <InputGroup>
       <InputGroupItem>
         <TextInput
           customIcon={<SearchIcon />}
-          aria-label={t('curiosity-toolbar.placeholder_filter', { context: 'displayName' })}
+          aria-label={t('curiosity-toolbar.placeholder', { context: ['filter', 'displayName'] })}
           className="curiosity-input__display-name"
           maxLength={255}
           onClear={onClear}
           onKeyUp={onKeyUp}
           value={currentValue}
-          placeholder={t('curiosity-toolbar.placeholder_filter', { context: 'displayName' })}
+          placeholder={t('curiosity-toolbar.placeholder', { context: ['filter', 'displayName'] })}
           data-test="toolbarFieldDisplayName"
         />
       </InputGroupItem>
     </InputGroup>
   );
 };
 
-/**
- * Prop types.
- *
- * @type {{useProduct: Function, t: translate, useDispatch: Function, useProductInventoryHostsQuery: Function}}
- */
-ToolbarFieldDisplayName.propTypes = {
-  t: PropTypes.func,
-  useDispatch: PropTypes.func,
-  useProduct: PropTypes.func,
-  useProductInventoryHostsQuery: PropTypes.func
-};
-
-/**
- * Default props.
- *
- * @type {{useProduct: Function, t: translate, useDispatch: Function, useProductInventoryHostsQuery: Function}}
- */
-ToolbarFieldDisplayName.defaultProps = {
-  t: translate,
-  useDispatch: storeHooks.reactRedux.useDispatch,
-  useProduct,
-  useProductInventoryHostsQuery
-};
-
-export { ToolbarFieldDisplayName as default, ToolbarFieldDisplayName };
+export { ToolbarFieldDisplayName as default, ToolbarFieldDisplayName, useOnClear, useOnKeyUp, useOnSubmit };
```

Comment:
required: 
the **additional mods to toolbarFieldDisplayName seem to be causing some testing inconsistencies** aka you didn't update the tests or you broke something else... so separate concerns... for this PR limit the changes to removal of deprecated react. Then in a subsequent PR implement the desired refactors